### PR TITLE
fix: Store AWS config per account

### LINKED ISSUE #13389: bug: AWS plugin syncing only last account (v22.7.0+)
Reported by `G4m3` on Discord:

> hi team, I have an issue with the latest version of the aws source plugin. The aws_account_contacts table gets the full_name (and potentially other fields) not reflecting the value of each account_id (ie. I have 10 accounts_id and the full_name is always the one from the first account). Is this  a regression ? Thank you!

Config:

```yaml
kind: source
spec:
  name: aws_dev
  path: cloudquery/aws
  version: "v22.7.0"
  tables: ["*"]
  skip_tables:
    - aws_ec2_vpc_endpoint_services
    - aws_cloudtrail_events
    - aws_docdb_cluster_parameter_groups
    - aws_docdb_engine_versions
    - aws_ec2_instance_types
    - aws_elasticache_engine_versions
    - aws_elasticache_parameter_groups
    - aws_elasticache_reserved_cache_nodes_offerings
    - aws_elasticache_service_updates
    - aws_iam_group_last_accessed_details
    - aws_iam_policy_last_accessed_details
    - aws_iam_role_last_accessed_details
    - aws_iam_user_last_accessed_details
    - aws_neptune_cluster_parameter_groups
    - aws_neptune_db_parameter_groups
    - aws_rds_cluster_parameter_groups
    - aws_rds_db_parameter_groups
    - aws_rds_engine_versions
    - aws_servicequotas_services
    - aws_ecr_repository_image_scan_findings
    - aws_athena_work_group_query_executions
  destinations: ["postgresql"]

  spec:
    accounts:
      - id: "dev"
        role_arn: "arn:aws:iam::xxxxxxxxxxx:role/xxxxxxxxxxxxx"
      - id: "dev2"
        role_arn: "arn:aws:iam::xxxxxxxxxxx:role/xxxxxxxxxxxxx"

kind: destination
spec:
  name: postgresql
  path: cloudquery/postgresql
  version: "v6.0.1"
  migrate_mode: forced

  write_mode: "overwrite-delete-stale"

  spec:
    connection_string: "postgresql://cloudquery:xxxxxxxxxx@localhost:5432/cloudquery?sslmode=disable"

```

CLI v3.14.5

User confirmed that this behavior is not happening in AWS v22.6.0, so this seems like a regression.

### PR DESCRIPTION
**Pull request by candiduslynx**
Fixes #13389

### CODE REVIEW

File: plugins/source/aws/client/services.go [hunk 1]
```diff
@@ -255,8 +257,8 @@ type Services struct {
 	Xray                         services.XrayClient
 }
 
-func (s *Services) InitService(awsConfig *aws.Config, service AWSServiceName) {
-	c := awsConfig.Copy()
+func (s *Services) InitService(service AWSServiceName) {
+	c := s.config
```
Comment (hermanschaaf) [inline plugins/source/aws/client/services.go:261]: To be on the safe side, can we keep the `Copy()` in here for now? I don't know if any of these `NewFromConfig` functions might actually mutate it.

Comment (candiduslynx) [reply plugins/source/aws/client/services.go]: 2b5d1ac

**Review by bbernays — APPROVED** [review]

File: plugins/source/aws/resources/services/timestream/databases.go [hunk 2]
```diff
@@ -46,10 +46,11 @@ func Databases() *schema.Table {
 
 func fetchTimestreamDatabases(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource, res chan<- any) error {
 	cl := meta.(*client.Client)
-	svc := cl.Services(client.AWSServiceTimestreamwrite).Timestreamwrite
+	services := cl.Services(client.AWSServiceTimestreamwrite)
+	svc := services.Timestreamwrite
 	// This should be removed once https://github.com/aws/aws-sdk-go-v2/issues/2163 is fixed
-	if cl.AWSConfig != nil && cl.AWSConfig.Region != cl.Region {
-		awsCfg := cl.AWSConfig.Copy()
+	if len(services.AWSConfig.Region) > 0 && services.AWSConfig.Region != cl.Region {
```
Comment (bbernays) [inline plugins/source/aws/resources/services/timestream/databases.go:52]: I think the better fix here might be to set the `Region` in the config in the test helper

**Review by bbernays — CHANGES_REQUESTED** [review]

File: plugins/source/aws/resources/services/timestream/databases.go (hunk 2 above)
Comment (candiduslynx) [reply plugins/source/aws/resources/services/timestream/databases.go]: 2e69feb

File: plugins/source/aws/resources/services/timestream/databases_mock_test.go [hunk 3]
```diff
@@ -32,7 +33,10 @@ func buildTimestreamDatabasesMock(t *testing.T, ctrl *gomock.Controller) client.
 	m.EXPECT().ListTagsForResource(gomock.Any(), gomock.Any(), gomock.Any()).Return(
 		&timestreamwrite.ListTagsForResourceOutput{Tags: tags}, nil)
 
-	return client.Services{Timestreamwrite: m}
+	return client.Services{
+		AWSConfig:       aws.Config{Region: "us-east-1"},
+		Timestreamwrite: m,
+	}
```
Comment (bbernays) [inline plugins/source/aws/resources/services/timestream/databases_mock_test.go:39]: That works or you could do this:

```
	services.Regions = []string{testOpts.Region}
	services.AWSConfig.Region = testOpts.Region
	c.accountMutex["testAccount"] = &sync.Mutex{}
```

In `AwsMockTestHelper`

Comment (candiduslynx) [reply plugins/source/aws/resources/services/timestream/databases_mock_test.go]: 921261c4694dd98c0ff4a6316de89549736484a8

**Review by bbernays — APPROVED** [review]

**Review by hermanschaaf — APPROVED** [review]

**Pull request merged**